raft.go

```diff
@@ -1667,9 +1667,13 @@ func (r *raft) handleAppendEntries(m pb.Message) {
 		hintIndex := min(m.Index, r.raftLog.lastIndex())
 		hintIndex = r.raftLog.findConflictByTerm(hintIndex, m.LogTerm)
 		hintTerm, err := r.raftLog.term(hintIndex)
-		if err != nil {
-			panic(fmt.Sprintf("term(%d) must be valid, but got %v", hintIndex, err))
+		if err != nil { // can be ErrCompacted or ErrUnavailable
```

Comment:
something like `panicUnlessErrCompactedOrErrUnavailable(err)`?

Reply:
Moved away from this error check.